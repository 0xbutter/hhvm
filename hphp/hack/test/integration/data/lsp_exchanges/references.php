<?hh

class C {
  private function ref_test_1 ($x) {  // Should not match
    $y = $x;           // Should not match
    static $x = 123;   // Should match
    $z = $x;           // Request ID 11: Looking for this $x // Should match
  }
}

function j(int $x): int {
  return $x;
}

function h(string $x): int {
  return 5;
}

function ref_test_2(int $x): void {   //  Should match
  $x = 3; // Request ID 12: Looking for this $x   //  Should match
  j($x) + $x + h("\$x = $x");         //  First, second, and fourth should match
  $lambda1 = $x ==> $x + 1;           //  Should not match
  $lambda2 = $a ==> $x + $a;          //  Should match
  $lambda3 = function($x) {           //  Should not match
    return $x + 1; };                 //  Should not match
  $lambda4 = function($b) use($x) {   //  Should match
    return $x + $b; };                //  Should match
}
