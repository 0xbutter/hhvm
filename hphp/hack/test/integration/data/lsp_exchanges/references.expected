[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "["
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 11,
    "result": [
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 5,
            "character": 11
          },
          "end": {
            "line": 5,
            "character": 13
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 6,
            "character": 9
          },
          "end": {
            "line": 6,
            "character": 11
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "result": [
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 9,
            "character": 18
          },
          "end": {
            "line": 9,
            "character": 33
          }
        }
      },
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 19,
            "character": 21
          },
          "end": {
            "line": 19,
            "character": 36
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 13,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 30,
            "character": 24
          },
          "end": {
            "line": 30,
            "character": 26
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 31,
            "character": 2
          },
          "end": {
            "line": 31,
            "character": 4
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 32,
            "character": 4
          },
          "end": {
            "line": 32,
            "character": 6
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 32,
            "character": 10
          },
          "end": {
            "line": 32,
            "character": 12
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 32,
            "character": 24
          },
          "end": {
            "line": 32,
            "character": 26
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 34,
            "character": 20
          },
          "end": {
            "line": 34,
            "character": 22
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 37,
            "character": 30
          },
          "end": {
            "line": 37,
            "character": 32
          }
        }
      },
      {
        "uri": "file://${root_path}${path_sep}references.php",
        "range": {
          "start": {
            "line": 38,
            "character": 11
          },
          "end": {
            "line": 38,
            "character": 13
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]